src/core/server/saved_objects/routes/export.ts

```diff
@@ -181,7 +181,12 @@ export const registerExportRoute = (
         });
       }
 
-      const exporter = context.core.savedObjects.exporter;
+      const includedHiddenTypes = supportedTypes.filter((supportedType) =>
+        typeRegistry.isHidden(supportedType)
+      );
+
+      const client = getClient({ includedHiddenTypes });
```

Comment:
It feels awkward having to extract the hidden types from the supported types to instantiate the client, but with current SO api it can't be avoided.